<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
	<meta name="keywords"
		content="custom BLAST, local BLAST server, BLAST web frontend, local BLAST, BLAST web interface, viroblast" />
	<title>Old site SequenceServer: BLAST search made easy!</title>

	<!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
	<!--[if lt IE 9]>
<script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

	<link type="text/css" rel="stylesheet" href="css/bootstrap.min.css" />
	<style type="text/css">
		section {
			padding-top: 60px;
		}

		div {
			text-align: justify;
		}

		section>.row {
			margin-bottom: 10px;
		}

		.content {
			padding: 40px 20px 40px 20px;
			background: white;
		}

		h2 {
			margin-bottom: 10px;
		}

		#about h2 {
			margin-bottom: 0px;
		}

		/*
//h1 { font-size: 3.8em; }
//margin-bottom: 0px;  margin-top:.5em;}

//pre {
//background: gray; color: #fff;
//padding: 15px; margin-left:
//15px; width: 650px;}

//.footer { text-align:center; padding-top:30px; font-style: italic; }
//img {max-width:80%; margin-left: auto; margin-right: auto;} */

		dd {
			padding-bottom: 10px;
		}

		.banner {
			display: block;
			max-width: 100%;
			padding: .5em 1em;
			margin-top: 1em;
			background-color: #ffffcc;
			color: inherit;
		}
	</style>

	<script>
		(function (i, s, o, g, r, a, m) {
			i['GoogleAnalyticsObject'] = r; i[r] = i[r] || function () {
				(i[r].q = i[r].q || []).push(arguments)
			}, i[r].l = 1 * new Date(); a = s.createElement(o),
				m = s.getElementsByTagName(o)[0]; a.async = 1; a.src = g; m.parentNode.insertBefore(a, m)
		})(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');

		ga('create', 'UA-471841-7', 'sequenceserver.com');
		ga('send', 'pageview');
	</script>
</head>

<body>

	<!-- Topbar ================================================== -->



	<div class="topbar" data-scrollspy="scrollspy">
		<div class="topbar-inner">
			<div class="container-fluid">
				<a class="brand" href="#">SequenceServer</a>
				<ul class="nav">
					<!--          <li><a href="#about">About</a></li> -->
					<!-- <li class="active"><a href="#about">About</a></li>   -->
					<li><a href="#installation">Installation</a></li>
					<li><a href="#license">License</a></li>
					<li><a href="#faq">FAQ</a></li>
					<li><a href="#support">Support</a></li>
					<li><a href="#contributors">Contributors</a></li>
					<li><a href="#users">Users/Citations</a></li>
					<li><a href="http://www.antgenomes.org/blast" title="Will open new window" target="_">&raquo; Demo
							Server &raquo;</a></li>
					<li><a href="https://github.com/yannickwurm/sequenceserver" title="Will open GitHub in new window"
							target="_">&raquo; Source Code &raquo;</a></li>
				</ul>
			</div>
		</div>
	</div>

	<div class="container-fluid" style="background:lightgray;">
		<div class="row">

			<div class="sidebar span4">
				<div style="padding-top:40px">
					<div style="width:20%; float:left; padding-top:10px;">
						<g:plusone size="medium" annotation="none" style="
  position: absolute; top: 20px; padding-top: 20px;"></g:plusone><br />
					</div>
					<div style="width:40%; float:left; padding-top:10px;">
						<a href="http://flattr.com/thing/397966/Simple-custom-local-BLAST-server" target="_blank">
							<img src="http://api.flattr.com/button/flattr-badge-large.png" alt="Flattr this"
								title="Flattr this" border="0" /></a>
					</div>
					<div style="width:40%; float:left; padding-top:5px;">
						<form action="https://www.paypal.com/cgi-bin/webscr" method="post">
							<input type="hidden" name="cmd" value="_s-xclick" />
							<input type="hidden" name="hosted_button_id" value="8G43L5PBKPSH8" />
							<input type="image" src="https://www.paypalobjects.com/en_US/i/btn/btn_donate_SM.gif"
								style="width:74px; height:21px; border: 0;" border="0" name="submit"
								alt="PayPal - The safer, easier way to pay online!" />
							<img alt="" border="0" src="https://www.paypalobjects.com/en_US/i/scr/pixel.gif" width="1"
								height="1" />
						</form>
					</div>
				</div>

				<div class="news small" style="text-align:justify;">
					<h4>News</h4>
					<p><strong>2013-2014</strong> Development is ongoing (check the
						<a href="https://github.com/yannickwurm/sequenceserver/commits/master">github
							log</a> or <a
							href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">mailing list</a>);
						contributors &amp;
						contributions always welcome!
					</p>
					<p><strong>2012-12</strong> Learn more or get help on <a
							href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">SequenceServer's
							Google Group</a> discussion forum.</p>
					<p><strong>2012-08</strong> SequenceServer cited: Elphick MR (2012) <a
							href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0044492">The
							Protein Precursors of Peptides That Affect the Mechanics of Connective Tissue and/or Muscle
							in the Echinoderm Apostichopus japonicus.</a> PLoS ONE.</p>
					<p><strong>2012-06</strong> SequenceServer 0.8.0 with loads of interface improvements released:
						automagic BLAST method selection, index for multi-query BLAST, sequence type feedback, to name a
						few.</p>
					<p><strong>2011-12</strong> SequenceServer needs your help. Contribute to a useful real world
						bioinformatics project by <a
							href="https://github.com/yannickwurm/sequenceserver/issues">squashing some bugs or improving
							features</a>. Great opportunity for CS or Webdesign students interested in Bioinformatics or
						Useability. In particular, our <a href="http://twitter.github.com/bootstrap/">twitter
							bootstrap</a> CSS interface has potential. Please get in touch.</p>
					<p><strong>Ongoing</strong>: Bugs being resolved &amp; improvements being made. Check the <a
							href="https://github.com/yannickwurm/sequenceserver">gitHub page</a> for recent activity.
					</p>
					<p><strong>2011-06-09</strong>: SequenceServer is now beta1.</p>
					<p><strong>2011-06-01</strong>: SequenceServer can use Rack &amp; Passenger to integrate existing
						Apache web servers.</p>
					<p><strong>2011-02-01</strong>: SequenceServer was deployed on the <a
							href="http://www.antgenomes.org">Fourmidable ant genome database</a> simultaneously with the
						publication of three <a href="http://www.pnas.org/cgi/doi/10.1073/pnas.1009690108"
							title="fire ant genome">ant genomes</a></p>
				</div>
			</div>
			<div class="content span11">
				<div role="banner" class="banner">
					<h1>This page is now deprecated.</h1>
					<p style="font-size:120%; text-align: left; line-height: 1.5;">
						For the most up to date information<a href="https://sequenceserver.com/"
							class="alert-link"> please visit the new homepage at
							https://sequenceserver.com/</a>
					</p>
				</div>
				<section id="about">
					<div class="page-header">
						<h1>SequenceServer&nbsp;<small>(BETA)</small></h1>
						<h2>BLAST search made easy!</h2>
					</div>
					<div class="row">
						<div class="span5" style="text-align: justify;">
							<p> SequenceServer lets you rapidly set up a BLAST+ server with an
								intuitive user interface for use locally or over the web.</p>

							<p>Sequence data is generated at increasing rates. However it is
								challenging to effectively share and query such data. Deploying complex
								solutions such as <a href="http://www.gmod.org">GMOD</a> is often overkill
								and <a href="http://www.ncbi.nlm.nih.gov/BLAST/docs/wwwblast.html">publicly available
									front ends for BLAST</a>
								are difficult to use or install.</p>

							<p>SequenceServer is an efficient and elegant alternative.
								Sequenceserver is still being developed yet is fully functional and has
								already been deployed for <a href="#users">multiple databases</a>.</p>

							<p>SequenceServer is <a href="#license">free
									for academics</a>. Its <a href="http://github.com/yannickwurm/sequenceserver">source
									code is
									available on GitHub</a>.</p>
						</div>
						<div class="span6">
							<ul class="media-grid">
								<li>
									<a href="http://www.antgenomes.org/blast">
										<img class="thumbnail" src="img/overview.png" style="width:100%; height: auto"
											alt="Ant genome BLAST
	server." />
									</a>
								</li>
							</ul>
						</div>
					</div>

					<h3>Easy for biologists</h3>
					<ul>
						<li>Minimalistic, clutter-free design, so you can focus on the biology</li>
						<li>The smart user interface automagically figures out the appropriate
							BLAST method to use based on your input and selected databases</li>
						<li>Common mistakes prevented. E.g., SequenceServer warns if you mix nucleotide and protein
							sequences</li>
						<li>Advanced users: use advanced parameters (e.g.
							<code>-task blastn-short -evalue 1.0e-20</code>) like you would in the command line.
						</li>
						<li>Simple overview of results</li>
						<li>Download links for hits</li>
					</ul>

					<h3>Rapidly deployed by admins</h3>
					<ul>
						<li>very little configuration required: only a directory
							containing formatted or un-formatted sequence databases</li>
						<li>SequenceServer scans the database directory during startup. And will interactively help you
							create BLAST databases from FASTA files</li>
						<li>Comes with a built-in web server. No need to deal with Apache or Nginx to run
							SequenceServer.</li>
						<li>Easily customizable hyperlinks to search hits (e.g. to your local genome browser or custom
							database)</li>
					</ul>

					<h3>Requirements</h3>

					<p><a href="http://ruby-lang.org">Ruby</a> (&gt;= 1.8.7), <a
							href="https://rubygems.org/">RubyGems</a> (&gt;= 1.3.6), and <a
							href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">NCBI
							BLAST+</a> (&gt;= 2.2.25+). That's it!</p>

					<p>Linux and MacOS are officially supported. While we would like to also support Windows,
						our resources are limited and we prefer to first concentrate on making SequenceServer great on
						fewer platforms.</p>

					<h3>Demo server</h3>

					<p>Sure, check <a href="http://www.antgenomes.org/blast">Ant genome BLAST</a> or the sites of <a
							href="#users">some of our users</a>.</p>
				</section>

				<!-- Setup ================================================== -->
				<section id="installation">
					<div class="page-header">
						<h1>Download &amp; Setup<small> in just a few minutes</small></h1>
					</div>
					<h2>Installation</h2>

					<p>Run the following from a command line to get the latest
						release of SequenceServer.</p>

					<pre> $ gem install sequenceserver</pre>

					<p>If that doesn't work, try <code>sudo gem</code> instead of
						<code>gem</code>. And subsequently follow the instructions on
						screen.
					</p>



					<h2>Configuration</h2>

					<p>SequenceServer needs to know the location of the BLAST+ binaries
						(<a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">NCBI
							BLAST+ download directory</a> or
						<code>brew tap homebrew/science; brew install blast</code>),
						and the BLAST databases. This is accomplished by providing a
						<code>.sequenceserver.conf</code> in the user's home directory.
					</p>

					<pre># .sequenceserver.conf
bin: ~/ncbi-blast-2.2.25+/bin/
database: /Users/me/blast_databases/
      </pre>

					<p>SequenceServer automatically generates a fully commented
						configuration file (<code>.sequenceserver.conf</code>) in the user's
						home directory if it can't find one. Uncomment the relevant lines
						(i.e remove the <code>#</code> signs), edit the values, and run
						SequenceServer.</p>

					<p>If you you already have databases, you're ready to <a href="#launch">launch SequenceServer</a>.
						If not, <a href="#dbs">set up databases as below</a>.</p>


					<h2 id='launch'>Launch SequenceServer</h2>

					<p>Simply execute:</p>
					<pre>$ sequenceserver </pre>

					<p>SequencerServer will check your config file and start up a
						webserver on <a href="http://localhost:4567/">http://localhost:4567/</a>. The
						port number can be changed in the config file. </p>
					<p>You're done. BLAST away.</p>




					<h2 id='dbs'>If needed: Set up custom BLAST databases</h2>

					<p>To set up BLAST databases from a directory of FASTA sequence
						files, use <code>sequenceserver</code>'s
						<code>format-databases</code> subcommand:
					</p>

					<pre>$ sequenceserver format-databases</pre>

					<p>It will automatically pick up the database directory from the
						configuration file. Or pass the database directory from command
						line:</p>

					<pre>$ sequenceserver format-databases directory_with_fasta_files</pre>

					<p>The script will assist you by:</p>

					<ul>
						<li>finding FASTA files in <code>directory_with_fasta_files</code> (and in subdirectories too)
						</li>
						<li>detecting the sequence type of each file</li>
						<li>asking you to name each database</li>
						<li>subsequently running the appropriate <code>makeblastdb</code> commands.</li>
					</ul>

					<p>Alternatively, manually set up BLAST databases from single FASTA
						sequence files. In the directory containing the FASTA file, run:</p>

					<pre>$ makeblastdb -dbtype &lt;db type&gt; -title &lt;db title&gt; -in &lt;db&gt; -parse_seqids</pre>

					<p>Where,</p>

					<ul>
						<li><code>&lt;db type&gt;</code> is either <code>prot</code>, or <code>nucl</code> depending on
							the type of sequence</li>
						<li><code>&lt;db title&gt;</code> is what users will see</li>
						<li><code>&lt;db&gt;</code> is the path to the FASTA file</li>
						<li><code>-parse_seqids</code> is required to generate links for downloading search
							hits (yes, it's a bit slow on large files).</li>
						<li>Additional options at <code>makeblastdb -help</code>.</li>
					</ul>

				</section>
				<section id="Optional">
					<div class="page-header">
						<h1 id='passenger'>Optional: <small>SequenceServer on Apache or Nginx with Phusion
								Passenger</small></h1>
					</div>
					<p>For most people, simply running the SequenceServer as above is more than enough. But if you need
						to deploy SequenceServer BLAST as part of a standard website,
						and have it show up as a normal webdirectory, SequenceServer can be
						deployed on Apache or Nginx with Phusion Passenger™ (a.k.a mod_rails
						or mod_rack).</p>

					<h4>Set up Phusion Passenger™</h4>

					<p>Install <code>passenger</code> gem:</p>

					<pre>gem install passenger</pre>

					<p>Install passenger module for Apache/Nginx module</p>

					<pre>
# for apache2
passenger-install-apache2-module

# for nginx
passenger-install-nginx-module
      </pre>

					<p>Follow instructions on screen. The installer will let you know of
						any missing software that is required to setup passenger, and also
						how to install them. Towards the end of installation, after the
						installer has finished building modules corresponding to your
						webserver, it will ask you to edit your webserver configuration files
						and some lines, so that your webserver knows how to load
						<code>passenger</code>.
					</p>

					<p>For Apache, you can do something like this:</p>

					<ol>
						<li>Create a new file called <code>passenger.load</code> in
							<code>/etc/apache2/mods-available</code>, and add the lines there.
						</li>
						<li>Run <code>a2enmod passenger</code>. This will ask you to restart apache2 with instructions
							on how to do the same. On my machine, I had to run <code>/etc/init.d/apache2 restart</code>
						<li>Running <code>touch tmp/restart.txt</code> may required when changing configuration.</li>
						</li>
					</ol>

					<h4>Deploying Sequence Server</h4>

					<p>Apache <strong>Root URI</strong> (eg:antgenomes.org or localhost).</p>

					<p></p>

					<div class="highlight">
						<pre>
<span class="nt">&lt;VirtualHost</span> <span class="s">*:80</span><span class="nt">&gt;</span>
<span class="nb">ServerName</span> localhost
<span class="nb">DocumentRoot</span> <span class="sx">/home/yeban/src/sequenceserver/public</span>
<span class="nt">&lt;Directory</span> <span class="s">/home/yeban/src/sequenceserver/public</span><span class="nt">&gt;</span>
        <span class="nb">Allow</span> from <span class="k">all</span>
        <span class="nb">Options</span> -MultiViews
<span class="nt">&lt;/Directory&gt;</span>
<span class="nt">&lt;/VirtualHost&gt;</span>
        </pre>
					</div>


					<p>Apache <strong>Sub URI</strong> (e.g.: antgenomes.org/blast or
						localhost/sequenceserver).</p>

					<p></p>

					<div class="highlight">
						<pre><span class="nt">&lt;VirtualHost</span> <span class="s">*:80</span><span class="nt">&gt;</span>
<span class="nb">ServerName</span> localhost
<span class="nb">DocumentRoot</span> <span class="sx">/var/www/</span>
<span class="nt">&lt;Directory</span> <span class="s">/var/www</span><span class="nt">&gt;</span>
<span class="nb">Allow</span> from <span class="k">all</span>
<span class="nt">&lt;/Directory&gt;</span>

<span class="nb">RackBaseURI</span> <span class="sx">/sequenceserver</span>
<span class="nt">&lt;Directory</span> <span class="s">/var/www/sequenceserver</span><span class="nt">&gt;</span>
<span class="nb">Options</span> -MultiViews
<span class="nt">&lt;/Directory&gt;</span>
<span class="nt">&lt;/VirtualHost&gt;</span>
</pre>
					</div>
				</section>

				<!--Support ================================================== -->
				<section id="support">
					<div class="page-header">
						<h1>Support <small>Mailing List</small></h1>
					</div>

					<p>Have an issue in deploying SequenceServer? Something is not working
						as expected? Have a tip? A feature request? Or just want to encourage further development?
						Post it to <a
							href="https://groups.google.com/forum/?fromgroups#!forum/sequenceserver">SequenceServer
							Google Group</a> and we will work something out.</p>
					<p>If you are completely overwhelmed and need more support that the
						mailing list can provide, we are available for consulting which can
						range from custom support to server deployment and administration to
						implementing specific features.</p>


					<!--        <iframe id="forum_embed"
src="javascript:void(0)"
scrolling="no"
frameborder="0"
width="630"
height="510">
</iframe>

<script type="text/javascript">

document.getElementById('forum_embed').src =
'https://groups.google.com/forum/embed/?place=forum/sequenceserver'
+ '&showsearch=false&showtabs=false&hideforumtitle=true';
</script>

-->
				</section>

				<!--LICENSE ================================================== -->
				<section id="license">
					<div class="page-header">
						<h1>License</h1>
					</div>

					<h2>For academics and non-for-profit organizations</h2>
					SequenceServer is free &amp; you can do most things you want with
					it. Our <a
						href="https://github.com/yannickwurm/sequenceserver/blob/master/LICENSE.txt">License.txt</a>
					file details some constraints regarding use and re-distribution (e.g. that instances you install
					retain
					links to this website). Please remember to cite the software.


					<h2>For Commercial &amp; for-profit organizations</h2>
					<a href="forprofit.html">If you're a for-profit you can try
						it with no constraints for one month. Subsequently you must purchase
						a license (it's cheap, quick &amp; easy!)</a>.

					<p>Custom licensing is also possible. Get in touch.</p>
				</section>

				<!--FAQ ================================================== -->
				<section id="faq">
					<div class="page-header">
						<h1>FAQ <small>Frequently Asked Questions</small></h1>
					</div>

					<dl>
						<dt>How do I cite SequenceServer?</dt>
						<dd>A manuscript describing SequenceServer is <em>in prep</em>. In the mean time,
							please cite the following reference: "Priyam A., Woodcroft B.J., Wurm Y <em>in prep</em>.
							<strong>SequenceServer: BLAST searching made easy</strong>". <a href="#users">This has
								already been done several times.</a>
						</dd>

						<dt>What are the alternatives to SequenceServer?</dt>
						<dd>There are several. But they are suboptimal because they look like
							they're from the last millenium. And installing them is
							challenging.</dd>

						<dd>
							<ul>
								<li><a href="http://www.ncbi.nlm.nih.gov/BLAST/docs/wwwblast.html">NCBI's wwwblast</a>
									hasn't been maintained in years <a href="http://openwetware.org/wiki/Wwwblast">and
										is difficult to set up</a>.</li>
								<li><a
										href="http://bioinformatics.oxfordjournals.org/content/23/17/2334.full">ViroBLAST</a>
									requires Apache &amp; MySQL.</li>
								<li><a href="http://gmod.org/wiki/BLAST_Graphic_Viewer">GMOD's BLAST Graphic Viewer</a>
									hasn't been maintained in years.</li>
								<li><a href="http://www.blaststation.com">BLASTstation</a> is only for local BLAST.</li>
							</ul>
						</dd>

						<dt>Is there a demo server?</dt>
						<dd>SequenceServer provides <a href="http://www.antgenomes.org/blast">BLAST for ant genomes</a>.
							Some of our users also <a href="#users">provide publicly accessible servers</a>.</dd>

						<dt>What is a BLAST database? Can I make a custom BLAST database?</dt>
						<dd>The BLAST program doesn't understand FASTA files. But BLAST
							includes the <code>makeblastdb</code> tool that reformats FASTA
							into the optimized BLAST-friendly format. Use <code>makeblastdb</code> or
							<code>sequenceserver format-databases</code> to create a custom BLAST database.
						</dd>

						<dt>Can I use a preformatted BLAST database? Can I use alias?</dt>
						<dd>Preformatted BLAST databases can also be used with SequenceServer. Most preformatted
							database ar split between multiple files (like .00, .01...). Thanks to <a
								href="http://acgt.tumblr.com/">Mark Anthony Gibbins'</a> contributions, SequenceServer
							can correctly understand these.</dd>
						<dd>If you use a custom <code>.nal</code> or <code>.pal</code> BLAST database alias file, and
							copy this alias file as well as the linked databases into SequenceServer's <code>db</code>
							directory, SequenceServer will display both the alias and the linked databases. If you want
							to display only the alias, remove the linked databases from <code>db</code>, then edit the
							<code>DBLIST</code> line in the <code>.pal</code> or <code>.nal</code> file so that it
							contains the complete path of each of the database files (e.g:
							<code>DBLIST species1_custom_blast_db species2_custom_blast_database</code> may become:
							<code>DBLIST /Users/blastuser/mydbs/species1_custom_blast_db /Users/blastuser/mydbs/species2_custom_blast_database</code>)
						</dd>

						<dt>How does BLAST identify similar regions? Why don't the
							ends match? What is the bitscore? What is the e-value that BLAST
							returns? What do these numbers mean?</dt>
						<dd>Many detailed explanations exist egarding such questions, including <a
								href="http://www.ncbi.nlm.nih.gov/books/NBK21097/">at NCBI</a>
							and in the original BLAST articles. Here's our executive
							summary.</dd>
						<dd>BLAST is a heuristic, ie. it is fast and approximate instead
							of being slow and perfect. It starts by looking for a minimal 100% match (e.g. 11
							consecutive nucleotides with 100% identity between your query and
							the database sequence). It is done if it finds none (thus if
							exactly every 10th base is different, BLAST finds no results). If it does find a match, it
							extends that
							in both directions: identical (or
							similar) bases add points; differences are negative points. If too many points are lost, it
							stops
							aligning. It might not stop at the exact best place, thus ends
							might not match. The <a
								href="http://www.ncbi.nlm.nih.gov/books/NBK21106/def-item/app8/">"bitscore"</a> is the
							total number of points for the aligning
							region. The bigger it is, the stronger the
							alignment. But the bitscore doesn't take into account sequence
							length nor database size. The E-value does. It is better to
							look at E-values than bitscores. The E-value represents the number of times the
							observed alignment would be expected to occur by chance (it is
							not a p-value!);
							depends on the bitscore, the length of the query
							sequence, and the cumulative length of all sequences in the
							database. Its easier to talk about "strong" E-values
							(e.g. 1e-100 = 10<sup>-100</sup> = almost zero; impossible to obtain by chance) vs "weak"
							E-evalues (e.g
							0.1; for similarity that may be due to chance) that small vs
							large (which is always a bit confusing).</dd>

						<dt>What is the difference between BLAST and BLAST+? What about WU-BLAST or AB-BLAST?</dt>
						<dd><a href="publications/ncbiBlast.pdf">The initial BLAST</a> has been rewritten several times
							- most recently by NCBI as BLAST+. NCBI now use and recommend using BLAST+. <a
								href="publications/blastplus.pdf">The BLAST+ publication</a> explains why BLAST+ is
							easier to use &amp; faster than the old "legacy" BLAST. WU-BLAST is now commerical and
							called <a href="http://blast.advbiocomp.com/">AB-BLAST</a>. There is probably no good reason
							to use either. Note that the output formats change slightly from one BLAST implementation to
							the next.</dd>
						<dd>NCBI's BLAST+ is most actively developed, just use it. SequenceServer only supports BLAST+.
						</dd>

						<dt>I swear I installed BLAST but SequenceServer can't find it!</dt>
						<dd>Make sure you downloaded NCBI's Blast+ (Blast Plus!) and not legacy BLAST. The old legacy
							BLAST had programs such as <code>blastall</code> and <code>formatdb</code>. The programs in
							BLAST+ have names like <code>makeblastdb</code>, <code>blastn</code>,
							<code>tblastx</code>...
						</dd>
						<dd>Also, check that your SequenceServer configuration file points at the <code>bin</code>
							directory, e.g.: <code>/Users/stevejobs/software/ncbi-blast-2.2.25+/bin/</code></dd>

						<dt>Why doesn't sequence retrieval work?</dt>
						<dd>Probably because you forgot the <code>-parse_seqids</code> argument when running
							<code>makeblastdb</code> (or they were formatted using legacy <code>formatdb</code>). Blast+
							2.2.24 and later indicate that when <code>-parse_seqids</code> was not used by leaving a
							space between <code>&gt;</code> and hit identifiers (eg: <code>&gt; db_sequence001</code>
							instead of <code>&gt;db_sequence001</code>).
						</dd>

						<dt>How can I add custom links to hits?</dt>
						<dd>Easy. Edit the <code>customisation.rb</code> file and read the examples it provides. To add
							a single link, uncomment and edit the <code>construct_custom_sequence_hyperlink</code>
							method; to add multiple links edit the
							<code>construct_custom_sequence_hyperlinking_line</code> method.
						</dd>

						<dt>How do I tell SequenceServer to read a different configuration file?</dt>
						<dd>By using <code>--config</code> command line switch. For example:
							<p></p>
							<pre>$ sequenceserver --config sequenceserver_ants.conf</pre>
							You will need to set it in <code>config.ru</code> for Rack based deployment:
							<p></p>
							<pre>SequenceServer::App.config_file = 'sequenceserver_ants.conf'</pre>
							The above line should be added before
							<code>SequenceServer::App.init</code>, so the <code>config.ru</code>
							looks something like this:
							<p></p>
							<pre>
require 'rubygems'
require 'bundler/setup'
require 'sequenceserver'

SequenceServer::App.config_file = '/absolute/path/to/configuration_file'
SequenceServer::App.init
run SequenceServer::App
</pre>
						</dd>

						<dt>How do I change the server port?</dt>
						<dd>SequenceServer will by default be accessible on port 4567
							at <a href="http://localhost:4567/">http://localhost:4567/</a>.Change
							the port by editing <code>~/.sequenceserver.conf</code>.
						</dd>

						<dt>Can my colleagues access the server I set up on my computer?</dt>
						<dd>Yes, <strong>if they can access your
								machine</strong>. This usually requires being in the same
							subnetwork, or asking IT services to open your machine to the
							outside world. You'll have to tell your colleagues your IP
							address (find it in the Network or Sharing section of System
							Preferences) and the port. The complete URL they will have to
							use might be http://192.168.134.13:4567/ or
							http://myblastserver.local:4567/
						</dd>

						<dt>Do I need to install Apache?</dt>
						<dd>SequenceServer automatically uses ruby's internal web servers. No
							need for Apache. By default it uses <a
								href="http://www.ruby-doc.org/stdlib/libdoc/webrick/rdoc/index.html">Webrick</a>.
							Installling <a href="http://code.macournoyer.com/thin/">Thin</a>
							(<code>gem install thin</code>), or <a href="http://rubygems.org/gems/mongrel">Mongrel</a>
							(<code>gem install mongrel</code>)
							can improve performance. SequenceServer will automatically
							use these if installed.
						</dd>

						<dt>Can I integrate SequenceServer into an Apache or Nginx server?</dt>
						<dd>Yes. <a href="passenger">We provide instructions
								above</a>.
						</dd>

						<dt>Should I activate hyper-threading to give BLAST more virtual cores?</dt>
						<dd>Yes.</dd>

						<dt>Which technologies does SequenceServer use?</dt>
						<dd>SequenceServer uses <a
								href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">NCBI
								BLAST+</a>, <a href="http://www.sinatrarb.com/">Sinatra</a>, and
							<a href="http://jquery.com/">jQuery</a>.
						</dd>

						<dt>Can I have the source code? Can I contribute?</dt>
						<dd>SequenceServer's <a href="http://github.com/yannickwurm/sequenceserver">source code is on
								GitHub</a>. Downloading,
							forking, pulling, filing issues and contributing is most welcome!</dd>


						<dt>You say SequenceServer is easy to set up. But I'm not a computer person and I don't
							understand a thing. This is too hard!</dt>
						<dd>We are considering offering hosted (password-protected) servers for a fee (ie you upload the
							data; we set things up as you need them). Please get in touch by email if you are interested
							or need other consulting advice.</dd>


						<dt>Can I get a support contract which guarantees rapid response?</dt>
						<dd>Yes. Get in touch by email.</dd>

						<dt>Can SequenceServer do XXX? Why doesn't YYYY work? It's ugly, it crashes!?!?</dt>
						<dd>First, run <code>gem update</code> or <code>sudo gem update</code>. </dd>
						<dd> Feel free
							to add comments, vote, report bugs, request features
							and contribute code. All contributions, patches and
							pulls are welcome. Our <a href="https://github.com/yannickwurm/sequenceserver/issues">issues
								tracker</a> logs development activities. If you're not a coder, please
							consider supporting development by a
							donation. Donations can dramatically increase the
							speed at which new features are added!</dd>
					</dl>
				</section>

				<!--Contributors ================================================== -->
				<section id="contributors">
					<div class="page-header">
						<h1>Contributors</h1>
					</div>
					<dl>
						<dt><a href="http://yeban.in">Anurag Priyam</a></dt>
						<dd> is a software developer primarily interested in applied computing and web infrastructure
							development. He is currently pursuing his undergraduate studies at the Indian Institute of
							Technology, Kharagpur.</dd>
						<dt><a href="http://ecogenomic.org/users/ben-woodcroft">Ben J Woodcroft</a></dt>
						<dd> is researching microbial responses to permafrost thaw (climate change induced), and the
							bioinformatics of metagenomics. He is a post-doctoral fellow at the <a
								href="http://ecogenomic.org">Australia Centre for Ecogenomics</a>.</dd>
						<dt><a href="http://yannick.poulet.org">Yannick
								Wurm</a></dt>
						<dd> is an evolutionary biologist using genomics
							tools to study the evolution of ant societies (eg: <a
								href="http://yannick.poulet.org/publications/wurm2009antDatabase.pdf">antgenomes.org</a>,
							<a href="http://yannick.poulet.org/publications/wurm2011fireAntGenome.pdf">fire ant
								genome</a>, <a
								href="http://www.antgenomes.org/publications/wurm2010fireAntQueenDealationExpression.pdf">competing
								fire ant queens</a>). He is a lecturer at <a href="http://www.sbcs.qmul.ac.uk/">Queen
								Mary University London</a>.
						</dd>
						<dt>Maybe you?</dt>
						<dd>The SequenceServer team will happily accept contributions. Please get in touch to figure out
							where you could fit in. Cheers!</dd>
					</dl>
				</section>

				<!--Users ================================================== -->
				<section id="users">
					<div class="page-header">
						<h1>Users <small>include:</small></h1>
					</div>
					<!-- Get inspiration from backpackit.com/buzz
			  "Works like a charm" - Steffen Lemke @ Centre for Organismal Studies, Heidelberg
			  "this is filling a huge need in BLAST sequencing" - Taylor Ballou @ Benaroya Research Institute at Virginia Mason
			  "good for the biologists within the group as the standard fare from
			  wwwBLAST is seriously lacking" - guy leonard at nhm


			  <ul>
				<li>'Beta 1' is deployed by the <a href="http://www.antgenomes.org/">Keller lab</a> at <a href="http://www.unil.ch/central">University of
				Lausanne</a>; see <a href="http://www.antgenomes.org/blast">Fourmidable Sequences</a>.</li>
				<li><a href="https://github.com/yannickwurm/sequenceserver/issues">Guy Leonard @ National History Museum London</a> </li>
				<li>Steffen Lemke @ Center for Organismal Studies, Heidelberg.</li>
				<li>Researchers at <a href="http://www.benaroyaresearch.org">Benaroya Research Institute at Virginia Mason</a></li>
				<li>you... </li>
			  </ul>
			  -->

					<img src='img/logos/composition_of_logos.png' width="800" height="197"
						alt="SequenceServer has satisfied users at U Melbourne, Centre for Organismal Studies Heidelberg, Swiss Institute of Bioinformatics, IIT Karagpur, Natural History Museum London, Queen Mary University London, Benaroya Research Institute, University of Lausanne"
						style="max-width:100%; height:auto;" />
					<br /> <br />
					<p>As well as:</p>
					<ul>
						<li><a href="http://aviangenomes.org/blast-public">AvianGenomes.org</a></li>
						<li><a href="http://haroun.iplantc.org/sequenceserver">iPlantCollaborative</a> Saccharum Genome
							Database</li>
						<li><a href="http://fastdb.biocloud.ca/">BioCloud.ca</a></li>
						<li><a href="">The Streptocarpus rexii database</a> at <a href="http://beaconlab.it">
								Bioinformatics Evolution and COmparative Genomics lab, Uni Milan</a></li>
						<li><a href="http://culture.no-ip.org/blast.shtml">Dr. Wolfang Rumpf @ U Maryland</a></li>
						<li><a href="http://www.ecology.kyoto-u.ac.jp/">Center for Ecological Research, Kyoto
								University</a></li>
						<li><a href="http://morse-ccmar.ualg.pt/edge/">European clam database</a> @ CCMAR at University
							of Algarve</li>
						<li><a href="http://spottedwingflybase.oregonstate.edu/blast">Spotted Drosophila Flybase</a> @
							Oregon State</li>
						<li><a href="http://www.sviridis.com">Setariabase</a> for Setaria viridis at Donald Danforth
							Plant Science Center</li>
						<li><a href="https://geefu.oadb.tsl.ac.uk/sequenceserver"></a> at <a
								href="https://geefu.oadb.tsl.ac.uk">Open Ash Dieback GeeFu</a></li>
						<li><a href="http://amborella.uga.edu/blast/">Amborella
								Genome</a> at University of Georgia</li>
						<li><a href="http://jekely-lab.tuebingen.mpg.de/blast/">Jekely
								Lab</a> at Max Planck Institute for Developmental Biology, Tuebigen</li>
						<li>Universitat Greifswald (private)</li>
						<li>Department of Plant Biology and Forest Genetics, Swedish
							University of Agricultural Sciences (private)</li>
						<li>Norwich BioScience Institutes (private)</li>
						<li>Australia's Commonwealth Scientific and Industrial
							Research Organisation (private)</li>
						<li>Max Planck Institute of Molecular Cell Biology and
							Genetics (private)</li>
						<li>School of Biological Sciences, <a href="http://www.biology.uq.edu.au">University of
								Queensland</a> (private)</li>
					</ul>
					<p>and many additional private installations we don't know about...</p>
					</ul>
					<h1>Citations include:</h1>
					<ul>
						<li>Mondav R et al (2014) <a
								href="http://www.nature.com/ncomms/2014/140214/ncomms4212/full/ncomms4212.html">Discovery
								of a novel methanogen prevalent in thawing permafrost</a> Nature Communications 5, 3212.
						</li>
						<li>Rowe ML et al (2014) <a
								href="http://www.sciencedirect.com/science/article/pii/S0016648013004693">Neuropeptides
								and polypeptide hormones in echinoderms: New insights from analysis of the transcriptome
								of the sea cucumber <em>Apostichopus japonicus</em></a> General and Comparative
							Endocrinology 194, 43-55.</li>
						<li>Chiara M et al (2013) <a
								href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0080961"><em>De
									Novo</em> Assembly of the Transcriptome of the Non-Model Plant <em>Streptocarpus
									rexii</em> Employing a Novel Heuristic to Recover Locus-Specific Transcript
								Clusters</a> PLoS ONE.</li>
						<li>Semmens DC et al (2013) <a
								href="http://jeb.biologists.org/content/216/21/4047.full">Discovery of a novel
								neurophysin-associated neuropeptide that triggers cardiac stomach contraction and
								retraction in starfish</a> Journal of Experimental Biology 216, 4047-4053.</li>
						<li>Chiu JC et al (2013) <a
								href="http://www.g3journal.org/content/early/2013/10/14/g3.113.008185.short">Genome of
								<em>Drosophila suzukii</em>, the Spotted Wing <em>Drosophila</em></a> G3 g3.113.008185
						</li>
						<li>Elphick MR (2012) <a
								href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0044492">The
								Protein Precursors of Peptides That Affect the Mechanics of Connective Tissue and/or
								Muscle in the Echinoderm <em>Apostichopus japonicus</em>.</a> PLoS ONE.</li>
						<li>Shreve J et al (2013) <a
								href="http://www.sciencedirect.com/science/article/pii/S002219101200296X">A genome-wide
								survey of small interfering RNA and microRNA pathway genes in a galling insect.</a>
							Journal of Insect Physiology.</li>
						<li>Berlamino et al (2013) <a href="http://www.biomedcentral.com/1471-2105/14/S1/S2">SymGRASS: a
								database of sugarcane orthologous genes involved in arbuscular mycorrhiza and root
								nodule symbiosis</a>. BMC Bioinformatics.</li>
						<li>Elphick MR et al (2013) <a
								href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0059076">The
								Evolution and Diversity of SALMFamide Neuropeptides</a>. PLoS ONE.</li>
					</ul>

				</section>

				<section>
					<p>Parts of this software are &copy; 2011 University of Lausanne. </p>
					<p>Please cite: "Priyam A., Woodcroft B.J., Wurm Y <em>in prep</em>. <strong>SequenceServer: BLAST
							searching made easy</strong> <code>http://sequenceserver.com</code>."
				</section>
			</div> <!-- /content -->
		</div>
	</div>
</body>

</html>
